ouroboros-network/wireshark-plugin/ouroboros_network.lua

```diff
@@ -101,72 +101,110 @@ ouroboros.fields = {
 	on_blockfetch_msg,
 	on_txsubmission_msg,
 	on_keepalive_msg,
-	on_peersharing_msg
+	on_peersharing_msg,
+	on_cbor_payload
 }
 
 
 function ouroboros.dissector(tvbuf, pktinfo, root)
 
 	local pktlen = tvbuf:len()
 	local offset = 0
+	local total = 0
 
 	while offset < pktlen do
 		result = dissectOuroboros(tvbuf, pktinfo, root, offset)
 		if result > 0 then
 			-- We parsed an Ouroboros message
 			offset = offset + result
-		else
-			-- we need more bytes
+			total = total + result
+		elseif result < 0 then
+			-- we need more bytes for current frame
 			pktinfo.desegment_offset = offset
-			result = -result
-			pktinfo.desegment_len = result
-			return pktlen
+			pktinfo.desegment_len = -result
+			return -result
+		else
+			-- some error
+			break
 		end
 	end
+
+	return total
 end
 
 dissectOuroboros = function (tvbuf, pktinfo, root, offset)
 
-	local msglen = tvbuf:len() - offset
+	local pktlen = tvbuf:len()
+	local msglen = pktlen - offset
+
 	if msglen < ON_HDR_LEN then
 		-- Not enough data to read the header, we need at least one more segmet.
-		return -DESEGMENT_ONE_MORE_SEGMENT
+		local need = ON_HDR_LEN - msglen
+		return -need
```

Comment:
Can we dissect if a mini-protocol message was split into multiple segments? :thinking: there's also mux in the way...